registration/api/v2/views.py

```diff
@@ -258,17 +213,83 @@ def get_queryset(self):
+            return BetaTestApplication.objects.all()
         else:
             return BetaTestApplication.objects.filter(user=self.request.user)
 
     def get_serializer_class(self):
-        if self.action == 'delete':
-            return {}
+        if self.action == "create":
+            return OpenEdXInstanceDeploymentCreateSerializer
 
-        return OpenEdXInstanceDeploymentSerializer
+        return OpenEdXInstanceDeploymentStatusSerializer
+
+    @swagger_auto_schema(
+        responses={**VALIDATION_AND_AUTH_RESPONSES, 200: openapi.Response("Changes committed"), },
+        manual_parameters=[
+            openapi.Parameter(
+                "force",
+                openapi.IN_QUERY,
+                type=openapi.TYPE_BOOLEAN,
+                description="Force launching a new instance even if one is already in progress.",
+            )
+        ],
+    )
+    def create(self, request, *args, **kwargs):
+        """
+        Commit configuration changes to instance and launch new AppServer.
+
+        This API call will copy over any changes made to the instance config to
+        the actual instance used to launch AppServers and launch a new AppServer
+        with the applied changes.
+
+        It checks if an AppServer is already being provisioned and in that
+        case prevents a new one from being launched unless forced.
+        """
+        force = request.query_params.get("force", False)
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        try:
+            instance_config = BetaTestApplication.objects.get(id=serializer.data['id'])
+        except BetaTestApplication.DoesNotExist:
+            raise ValidationError(
+                "The specified instance was not found.", code="instance-not-found",
+            )
+
+        instance = instance_config.instance
+        if instance is None:
+            # For a new user an instance will not exist until they've verified their email.
+            raise ValidationError(
+                "Must verify email before launching an instance", code="email-unverified",
+            )
+
+        if not force and instance.get_provisioning_appservers().exists():
+            raise ValidationError("Instance launch already in progress", code="in-progress")
+
+        if not EmailAddress.objects.get(email=instance_config.public_contact_email).is_confirmed:
+            raise ValidationError(
+                "Updated public email needs to be confirmed.", code="email-unverified"
+            )
+
+        instance_config.commit_changes_to_instance(
+            spawn_on_commit=True,
+            retry_attempts=settings.SELF_SERVICE_SPAWN_RETRY_ATTEMPTS,
+        )
+
+        return Response(status=status.HTTP_200_OK)
+
+    def list(self, request, *args, **kwargs):
+        """
+        List method not allowed.
+        """
+        return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
```

Comment:
I think this is the default behaviour right? 
In any case, doesn't hurt to be explicit.

Reply:
The default behavior isn't implemented because I used only the exact views required (Create, Destroy, Retrieve). That meant that the abstract method `list` wasn't implemented, and this broke the browsable API. Adding this fixed it :)